lib/krikri/harvesters/couchdb_harvester.rb

```diff
@@ -0,0 +1,115 @@
+require 'couchrest'
+require 'mime/types'
+
+module CouchRest
+  class Streamer
+    ##
+    # CouchRest::Streamer relies on curl. Some CouchDB servers behind HTTPS
+    # may have self-signed certificates, so we're adding the `--insecure`
+    # parameter to ignore strict certificate checking. There is probably a
+    # more elegant way to do this. If we *do* need to monkey patch this, it
+    # probably should be done in Heidrun and not Krikri.
+    #
+    # Note that we'll probably have to do something similar for #get_record
+    # requests: https://github.com/rest-client/rest-client/issues/288
+    def initialize
+      self.default_curl_opts = [
+        '--silent',
+        '--insecure',
+        '--no-buffer',
+        '--tcp-nodelay',
+        '-H "Content-Type: application/json"'
+      ].join(' ')
+    end
+  end
+end
+
+module Krikri::Harvesters
+  ##
+  # A harvester implementation for CouchDB
+  class CouchdbHarvester
+    include Krikri::Harvester
+    attr_accessor :client
+
+    ##
+    # @param opts [Hash] options to pass through to client requests.
+    #   If {:couchdb => :view} is not specified, it defaults to using the
+    #   CouchDB `_all_docs` view.
+    # @see CouchRest::Database
+    # @see http://docs.couchdb.org/en/latest/api/database/bulk-api.html
+    #   CouchDB _all_docs endpoint
+    # @see http://docs.couchdb.org/en/latest/api/ddoc/views.html CouchDB views
+    # @see #expected_opts
+    def initialize(opts = {})
+      super
+      @opts = opts.fetch(:couchdb, view: '_all_docs')
+      @opts[:view] ||= '_all_docs'
+      @client = CouchRest.database(uri)
+    end
+
+    ##
+    # Streams a response from a CouchDB view to yield identifiers.
+    #
+    # The following will only send requests to the endpoint until it
+    # has 1000 record ids:
+    #
+    #     record_ids.take(1000)
```

Comment:
Probably, but I think `#lazy` is fine, here.  The streamer ends up downloading the whole view, anyway, if I understand correctly.